drkafka/src/main/java/com/pinterest/doctorkafka/KafkaClusterManager.java

```diff
@@ -599,9 +599,10 @@ public UnderReplicatedReason getUnderReplicatedReason(String brokerHost,
       List<OutOfSyncReplica> outOfSyncReplicas) {
     Map<TopicPartition, Integer[]> replicasMap = new HashMap<>();
     boolean success = true;
-
-    PriorityQueue<KafkaBroker> brokerQueue = kafkaCluster.getBrokerQueue();
+    PriorityQueue<KafkaBroker> brokerQueue;
     for (OutOfSyncReplica oosReplica : outOfSyncReplicas) {
+      // rebuild broker queue each time to get up-to-date results
+      brokerQueue = kafkaCluster.getBrokerQueue();
```

Comment:
`kafkaCluster.getBrokerQueue()` is an expensive operation.  As long as we maintain the priority queue ordering invariant, we do not need to recompute the priority queue for each out of sync replica.